### DISCUSSION TITLE
Julia LSP support

### DISCUSSION THREAD

**Discussion by AnuramAmarthya10** [SpartanJ/ecode#362] [Ideas]
Julia does have Language server. If would be a great feature to have

**Comment by SpartanJ** [comment]
I did not add it by default because they don't provide a single binary to run the server from, check [here](https://github.com/julia-vscode/LanguageServer.jl?tab=readme-ov-file#installation-and-usage):

```bash
julia --project=/path/to/LanguageServer.jl/environment \
  -e "using LanguageServer; runserver()" \
  <env_path>
```

since the language server path is unknown I can not add a generic configuration for all users. That's basically why I haven't add it. But you can add it yourself very easily, check [here](https://github.com/SpartanJ/ecode?tab=readme-ov-file#lsp-client).

**Reply by AnuramAmarthya10** [reply]
Thanks for the response! But JupyterLab automatically detects julia LanguageServer as mentioned [here.](https://discourse.julialang.org/t/is-it-possible-to-use-julias-languageserver-jl-with-jupyterlabs-jupyterlab-lsp-extension/40945/3)

**Reply by SpartanJ** [reply]
OK, I'll just assume that the LSP has been installed from the package as they seem to do. Added the LSP [here](https://github.com/SpartanJ/eepp/commit/1de8c60e75033ec2c148d17785c5575c3dc8b9d5).

**Comment by AnuramAmarthya10** [comment]
Editing the lspclient.json like this has worked for me.
```` json
  "servers": [
    {
      "command": "julia --project=C:\\Users\\Anuram\\.julia\\packages\\LanguageServer\\Fwm1f\\src\\LanguageServer.jl -e 'using LanguageServer; runserver()'",
      "file_patterns": [
        "*.jl$"
      ],
      "language": "julia",
      "name": "julia LanguageServer"
    }
  ]
